lib/Solver/QueryLoggingSolver.cpp

```diff
@@ -53,9 +59,15 @@ void QueryLoggingSolver::startQuery(const Query& query, const char* typeName,
               << "Instructions: " << instructions << "\n";
     
     printQuery(query, falseQuery, objects);
-    
+
+    if (DumpPartialQueryiesEarly) {
+      logBuffer.flush();
```

Comment:
@MartinNowack You probably should be calling ``QueryLoggingSolver::flushBuffer()`` here to avoid duplicating code. That method will probably need some refactoring so that it is usable though.